content/tutorials/build-apps/build-a-desktop-wallet-in-js.md

```diff
@@ -0,0 +1,241 @@
+
+## Prerequisites
+
+To implement this tutorial you should have a basic understanding of JavaScript and Node.js. You should also have a basic idea about XRP Ledger. For more information, visit the [XRP Ledger Dev Portal](https://xrpl.org) and the [XRPL Learning Portal](https://learn.xrpl.org/) for videos, libraries, and other resources.
+
+Follow the steps below to get started:
+
+1. [Node.js](https://nodejs.org/en/download/) (v10.15.3 or higher)
+2. Install [Yarn](https://yarnpkg.com/en/docs/install) (v1.17.3 or higher) or [NPM](https://www.npmjs.com/get-npm) (v6.4.1 or higher)
+
+## Source Code
+
+You can find the complete source code for all of this tutorial's examples in the [code samples section of this website's repository]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/).
+
+## Goals
+
+At the end of this tutorial, you should be able to build a simple XRP wallet displayed below.
+
+![Home Page Screenshot](img/js-wallet-home.png)
+
+This application is capable of the following:
+
+- Shows updates to the XRP Ledger in real-time.
+- Can view any XRP Ledger account's activity "read-only" including showing how much XRP was delivered by each transaction.
+- Shows how much XRP is set aside for the account's reserve requirement. 
+- Can send direct XRP payments, and provides feedback about the intended destination address, including:
+  - Displays available balance in your account
+  - Verifies that the destination address is valid
+  - Validates amount input to ensure it is a valid number and that the account has enough XRP to send
+  - Allows addition of the destination tag
+
+## Steps
+
+Before you begin, make sure you have the pre-requisites installed. Check your node version by running `node -v`. If you don't have node installed, you can download it [here](https://nodejs.org/en/download/).
+
+### Step 1: Setting up the project
+
+- Navigate to the directory that you want to create the project in.
+- Run `yarn create vite-app <project-name>` or `npm init vite-app <project-name>` to create a new project.
+- Install the dependencies that are mentioned below:
+
+```json
+{
+    "@esbuild-plugins/node-globals-polyfill": "^0.2.3",
+    "crypto-browserify": "^3.12.0",
+    "events": "^3.3.0",
+    "https-browserify": "^1.0.0",
+    "rollup-plugin-polyfill-node": "^0.12.0",
+    "stream-browserify": "^3.0.0",
+    "stream-http": "^3.2.0",
+    "vite": "^4.1.0",
+    "dotenv": "^16.0.3",
+    "xrpl": "^2.7.0-beta.2",
+    "xrpl-tagged-address-codec": "^0.2.1"
+}
+```
+you can install them by copying the dependencies from the [package.json](_code-samples/build-a-wallet/js/package.json) file to your project's package.json and running `yarn` or `npm i`. You can also do `yarn add <package-name>` or `npm install <package-name>` for each individual package.
```

Comment:
(Added numbers, but please make the numbers line up with the rest of the steps :) )
```suggestion
1. Copy this [package.json](_code-samples/build-a-wallet/js/package.json) file to your project.
        - Alternatively you can also do `yarn add <package-name>` or `npm install <package-name>` for each individual package to add them to your `package.json` file.
2. Run `yarn` or `npm i` to install the packages.
```